servers/src/main/java/tachyon/worker/block/meta/StorageDir.java

```diff
@@ -282,13 +282,14 @@ public void addBlockMeta(BlockMeta blockMeta) throws OutOfSpaceException, Alread
     long blockId = blockMeta.getBlockId();
     long blockSize = blockMeta.getBlockSize();
 
+    if (hasBlockMeta(blockId)) {
+      throw new AlreadyExistsException("Failed to add BlockMeta: blockId " + blockId + " exists");
+    }
```

Comment:
I wonder if it makes sense to have our own kind of preconditions, since we do a lot of this check condition and if true/false throw exception.